src/components/views/ViewDataTable/utils.tsx

```diff
@@ -42,25 +41,43 @@ export function makeGridColDef(viewCol: ZetkinViewColumn, orgId: number | string
     else if (viewCol.type == COLUMN_TYPE.PERSON_NOTES) {
         colDef.width = 300;
         colDef.renderCell = (params) => (
-            <PersonNotesViewCell params={ params as PersonNotesViewCellParams }/>
+            <PersonNotesViewCell params={ params as GridRenderCellParams }/>
         );
+        colDef.valueGetter = (params) => {
+            const notes = (params as PersonNotesParams).value;
+            return notes?.length ? notes.map(note => note.text).join(',') : null;
+        };
     }
     else if (viewCol.type == COLUMN_TYPE.LOCAL_PERSON) {
         colDef.minWidth = 50;
         colDef.renderCell = (params) => (
-            <LocalPersonViewCell orgId={ orgId } params={ params as LocalPersonViewCellParams }/>
+            <LocalPersonViewCell orgId={ orgId } params={ params as GridRenderCellParams }/>
         );
+        colDef.valueGetter = (params) => {
+            const person = (params as LocalPersonParams).value;
+            return person ? `${person.first_name} ${person.last_name}` : null;
+        };
     }
     else if (viewCol.type == COLUMN_TYPE.SURVEY_RESPONSE) {
         colDef.width = 300;
+        colDef.type = 'string';
         colDef.renderCell = (params) => (
-            <SurveyResponseViewCell params={ params as SurveyResponseViewCellParams }/>
+            <SurveyResponseViewCell params={ params as GridRenderCellParams }/>
         );
+        colDef.valueGetter = (params) => {
+            const responses = (params as SurveyResponseParams).value;
+            return responses?.length ? responses.map(response => response.text).join(',') : null;
```

Comment:
These should be ordered before joining, so that the response that's displayed in the GUI (the most recent one) is also the one being used for sorting (currently the first one in the array, which may not be most recent).

Reply:
I've filed a separate issue for this as #549